### PR TITLE
Move baked data lookup size calculations to its exporter

### PR DESCRIPTION
**Pull request by robertbastian**
This is more accurate, as the previous code needed to make some educated guesses. Needed for zerotrie where we can't guess.

Based on #5169

### CODE REVIEW

File: provider/data/datetime/fingerprints.csv [hunk 1]
```diff
@@ -1,6 +1,3 @@
-*, <'static strs>, 1478B, 382 unique static strings
-datetime/buddhist/datelengths@1, <lookup>, 4104B, 171 identifiers
-datetime/buddhist/datelengths@1, <total>, 22569B, 141 unique payloads
```
Comment (sffc) [inline provider/data/datetime/fingerprints.csv:3]: Issue: The stats are being deleted from the fingerprint file. I don't immediately see where the stats reappear. I see a comment in the .data.rs files?

Comment (robertbastian) [reply provider/data/datetime/fingerprints.csv:3]: Yes, a doc comment on the macro

**Review by Manishearth — DISMISSED** [review]

File: provider/data/datetime/fingerprints.csv (hunk 1 above)
Comment (robertbastian) [reply provider/data/datetime/fingerprints.csv:3]: will keep the totals

**Review by sffc — APPROVED** [review]

File: provider/data/datetime/fingerprints.csv (hunk 1 above)
Comment (sffc) [reply provider/data/datetime/fingerprints.csv:3]: Thank you!

**Review by sffc — DISMISSED** [review]

**Comment by robertbastian** [comment]
Rebased on https://github.com/unicode-org/icu4x/pull/5169

**Review by sffc — APPROVED** [review]
Suggestion (optional): add a space between the number and the "B" since it looks a lot like an "8" from a distance

**Pull request merged**